crates/sp-lightclient/src/lib.rs

```diff
@@ -86,6 +90,21 @@ pub struct ChainConstants<Header: HeaderT> {
 
     /// Slot probability.
     pub slot_probability: (u64, u64),
+
+    /// Eon duration at which next derived Salt is used.
+    pub eon_duration: u64,
+
+    /// Interval after the eon change when next salt is revealed
+    pub next_salt_reveal_interval: u64,
+}
+
+/// Data that is useful to derive the next salt.
+#[derive(Default, Debug, Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+pub struct SaltDerivationInfo {
+    /// Current eon index.
+    pub eon_index: EonIndex,
+    /// Randomness used to derive next salt
+    pub maybe_randomness: Option<Randomness>,
```

Comment:
Should randomness be optional? Maybe the whole data structure should be wrapped in `Option` instead?

Reply:
> Should randomness be optional?

We only need to track the randomness when the salt is revealed in a given eon. So this will be None From the time eon changed till salt is revealed and them carries the Randomness till the eon changes to next one.

> Maybe the whole data structure should be wrapped in Option instead?

We can but we would need to derive the current eon index by tracing the tree back to point when eon has changed. Tracking current eon index will make it much simpler.